util/test/testutil.go

```diff
@@ -196,19 +198,72 @@ func oidcMockHandler(t *testing.T, url string) func(http.ResponseWriter, *http.R
 			require.NoError(t, err)
 			_, err = w.Write(out)
 			require.NoError(t, err)
+		case "/token":
+			if assertAzureWorkloadIdentityAssertions {
+				err := r.ParseForm()
+				require.NoError(t, err)
+
+				formData := r.Form
+				clientAssertion := formData.Get("client_assertion")
+				clientAssertionType := formData.Get("client_assertion_type")
+				assert.Equal(t, "serviceAccountToken", clientAssertion)
+				assert.Equal(t, "urn:ietf:params:oauth:client-assertion-type:jwt-bearer", clientAssertionType)
+			}
+
+			response, err := mockTokenEndpointResponse(url)
+			require.NoError(t, err)
+			out, err := json.Marshal(response)
+			require.NoError(t, err)
+			_, err = w.Write(out)
+			require.NoError(t, err)
 		default:
 			w.WriteHeader(http.StatusNotFound)
 		}
 	}
 }
 
-func GetOIDCTestServer(t *testing.T) *httptest.Server {
+func GetOIDCTestServer(t *testing.T, assertAzureWorkloadIdentityAssertions bool) *httptest.Server {
```

Comment:
Use a method specific for the Azure OIDC server instead of parametrizing with a boolean.
```suggestion
func GetAzureOIDCTestServer(t *testing.T) *httptest.Server
```

Reply:
Done